segment.go

```diff
@@ -581,37 +587,62 @@ func (s *SegmentBase) Count() uint64 {
 	return s.numDocs
 }
 
+func (s *SegmentBase) getDocIDinfo() (*Dictionary, string, error) {
+	// obtain a read lock to check if the max doc ID and dict for _id is cached
+	s.docIDMutex.RLock()
+	cachedDocID := s.cachedMaxDocID
+	cachedDict := s.idDict
+	s.docIDMutex.RUnlock()
```

Comment:
This should move to line 596.5 and another unlock after the conditional at 599.5.